### PR TITLE
feat: add cpp solutions

### PR DESCRIPTION
**Pull request by sameer03697**
- cpp solutions for various DSA and Algo problems

### CODE REVIEW

**Comment by bhushan** [comment]
Hi @sameer03697 .. thank you for your PR. I have updated directory structure of the repo. Please create `cpp` folder and root and then put all your changes inside that folder. 

Things to be kept in mind while you refactor,
- we are using `snake-case` for files and directories
- try to segregate questions based on category same as we are doing for javascript solutions. for `cpp` we can have `input`, `output` and `solution.cpp` files for all solutions.
 `cpp` => `binary-tree` => `preorder-traversal` => here we can have these three files. something like this.
- also for commit messages i am following `[BinaryTree]: Kth sum problem` this kind of pattern, please use this pattern to keep things consistent.

File: dsaProblemsCPP/readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This is for DSA problems
```
Comment (bhushan) [inline dsaProblemsCPP/readme.md]: we dont need readme here

File: dsaProblemsCPP/binarySearch/KthSumValue/sol.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    for(int i=0;i<m;i++){
+        cin>>b[i];
+    }
+    if(m>n){
+    swap(a,b);
+    swap(n,m);
+    }
+    sort(a.begin(), a.end());
+    sort(b.begin(), b.end());
+    /*for(auto i: a){
```
Comment (bhushan) [inline dsaProblemsCPP/binarySearch/KthSumValue/sol.cpp]: remove unwanted code

File: dsaProblemsCPP/binarySearch/KthSumValue/sol.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+       } 
+    }
+    if(count < k){
+        return false;
+    }
+    //assert(1==0);
+    throw invalid_argument("b");
+}
+void solve(){
+    //cout<<"I am Sameer Kumar"<<endl;
```
Comment (bhushan) [inline dsaProblemsCPP/binarySearch/KthSumValue/sol.cpp]: remove commented code

**Review by bhushan — CHANGES_REQUESTED** [review]

**Comment by sameer03697** [comment]
Did a force push 😢 . Please do a review.

**Review by bhushan — CHANGES_REQUESTED** [review]
add readme.md with instructions to run programs and install utils. and also change file names to input.txt and solution.cpp

**Comment by sameer03697** [comment]
@bhushan Please review this PR

**Comment by sameer03697** [comment]
@bhushan Please review this PR.  I willl do Utils for cpp next time.

**Comment by sameer03697** [comment]
Added first 18 problems, please review this, I will add more problems soon.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+*
```
Comment (bhushan) [inline .gitignore:1]: please put this file inside cpp folder

**Pull request merged**